### PR TITLE
`azurerm_kubernetes_cluster` - `http_proxy_config` supports update

### PR DESCRIPTION
**Pull request by ms-henglu**
image

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
LGTM 🦕

File: internal/services/containers/kubernetes_cluster_network_resource_test.go [hunk 1]
```diff
@@ -813,6 +813,38 @@ func TestAccKubernetesCluster_httpProxyConfig(t *testing.T) {
...
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster", "test")
+	r := KubernetesClusterResource{}
+	noProxy := "\"localhost\", \"127.0.0.1\", \"mcr.microsoft.com\""
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.httpProxyConfig(data, noProxy),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+			ExpectNonEmptyPlan: true,
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_network_resource_test.go:826]: Could you explain why we need this here?

**Review by stephybun — CHANGES_REQUESTED** [review]

File: internal/services/containers/kubernetes_cluster_network_resource_test.go (hunk 1 above)
Comment (ms-henglu) [reply internal/services/containers/kubernetes_cluster_network_resource_test.go:826]: Hi @stephybun, sorry for late response.

There're tags added to the `azurerm_linux_virtual_machine` by some policies:
```
  ~ tags                                                   = {
                  - "azsecpack"                                                                = "nonprod" -> null
                  - "platformsettings.host_environment.service.platform_optedin_for_rootcerts" = "true" -> null
                }
```

I refers the following codes where it adds "ExpectNonEmptyPlan: true"

https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_network_resource_test.go#L799C2-L799C2

**Review by stephybun — APPROVED** [review]
Thanks for the clarification @ms-henglu. LGTM 👍

**Pull request merged**